src/overlays/actors/ovl_En_Baba/z_en_baba.c

```diff
@@ -311,357 +336,364 @@ void func_80BA8DF4(EnBaba* this, PlayState* play) {
+            this->animIndex = BABA_ANIM_IDLE_HOLDING_BAG;
+            Actor_ChangeAnimationByInfo(&this->skelAnime, sAnimations, this->animIndex);
+            this->actionFunc = EnBaba_FaceForward;
         }
     }
 }
 
-void func_80BA9758(EnBaba* this, PlayState* play) {
-    if ((this->unk_40A & 8) || (this->unk_144 != NULL) || func_80BA8820(this, play)) {
+void EnBaba_Idle(EnBaba* this, PlayState* play) {
+    if ((this->flags & ENBABA_AUTOTALK) || (this->bombShopkeeper != NULL) || EnBaba_FindBombShopkeeper(this, play)) {
         if (Actor_ProcessTalkRequest(&this->actor, &play->state)) {
-            func_80BA886C(this, play);
-            if (this->unk_40A & 8) {
+            EnBaba_HandleConversation(this, play);
+            if (this->flags & ENBABA_AUTOTALK) {
                 this->actor.flags &= ~ACTOR_FLAG_10000;
             }
-            this->actionFunc = func_80BA98EC;
+            this->actionFunc = EnBaba_Converse;
         } else if (this->actor.xzDistToPlayer < 100.0f) {
-            if (this->unk_40A & 8) {
+            if (this->flags & ENBABA_AUTOTALK) {
                 this->actor.flags |= ACTOR_FLAG_10000;
             }
             func_800B8614(&this->actor, play, 100.0f);
         }
     }
 }
 
-void func_80BA9848(EnBaba* this, PlayState* play) {
-    u8 temp_v0 = Message_GetState(&play->msgCtx);
+void EnBaba_FollowScheduleConverse(EnBaba* this, PlayState* play) {
+    u8 talkState = Message_GetState(&play->msgCtx);
 
-    if (((temp_v0 == 5) || (temp_v0 == 6)) && Message_ShouldAdvance(play)) {
+    if (((talkState == 5) || (talkState == 6)) && Message_ShouldAdvance(play)) {
         play->msgCtx.msgMode = 0x43;
         play->msgCtx.unk12023 = 4;
-        this->actionFunc = func_80BA9B80;
+        this->actionFunc = EnBaba_FollowSchedule;
     }
     Math_SmoothStepToS(&this->actor.shape.rot.y, this->actor.yawTowardsPlayer, 5, 0x1000, 0x100);
 }
 
-void func_80BA98EC(EnBaba* this, PlayState* play) {
-    u8 temp_v0 = Message_GetState(&play->msgCtx);
+void EnBaba_Converse(EnBaba* this, PlayState* play) {
+    u8 talkState = Message_GetState(&play->msgCtx);
 
-    if (temp_v0 == 5) {
+    if (talkState == 5) {
         if (Message_ShouldAdvance(play)) {
-            if (this->unk_40A & 1) {
-                this->unk_40A &= ~1;
+            if (this->flags & ENBABA_END_CONVERSATION) {
+                this->flags &= ~ENBABA_END_CONVERSATION;
                 play->msgCtx.msgMode = 0x43;
                 play->msgCtx.unk12023 = 4;
-                if (this->unk_40A & 8) {
+                if (this->flags & ENBABA_AUTOTALK) {
                     if (CHECK_QUEST_ITEM(QUEST_BOMBERS_NOTEBOOK)) {
                         if (play->msgCtx.unk120B1 == 0) {
                             gSaveContext.save.weekEventReg[81] |= 2;
-                            func_80BA8C4C(play, 0xD670);
+                            EnBaba_SetNextEntrance(play, 0xD670);
                             return;
```

Comment:
Is this return needed? Seems like the rest is just else and if-else cases

Reply:
Unfortunately it is required.